components/overflow-group/overflow-group-mixin.js

```diff
@@ -0,0 +1,349 @@
+import { css, html } from 'lit';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+import { offscreenStyles } from '../offscreen/offscreen.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const OVERFLOW_DROPDOWN_CLASS = 'd2l-overflow-dropdown';
+export const OVERFLOW_MINI_DROPDOWN_CLASS = 'd2l-overflow-dropdown-mini';
+
+const AUTO_SHOW_CLASS = 'd2l-button-group-show';
+const AUTO_NO_SHOW_CLASS = 'd2l-button-group-no-show';
+
+const OPENER_TYPE = {
+	DEFAULT: 'default',
+	ICON: 'icon'
+};
+
+async function filterAsync(arr, callback) {
+	const fail = Symbol();
+	const results = await Promise.all(arr.map(async item => {
+		const callbackResult = await callback(item);
+		return callbackResult ? item : fail;
+	}));
+	return results.filter(i => i !== fail);
+}
+
+export const OverflowGroupMixin = superclass => class extends LocalizeCoreElement(superclass) {
+
+	static get properties() {
+		return {
+			/**
+			 * Use predefined classes on slot elements to set min and max buttons to show
+			 * @type {boolean}
+			 */
+			autoShow: {
+				type: Boolean,
+				attribute: 'auto-show',
+			},
+			/**
+			 * @ignore
+			 */
+			chompIndex: {
+				type: Number,
+			},
+			/**
+			 * minimum amount of buttons to show
+			 * @type {number}
+			 */
+			minToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'min-to-show',
+			},
+			/**
+			 * maximum amount of buttons to show
+			 * @type {number}
+			 */
+			maxToShow: {
+				type: Number,
+				reflect: true,
+				attribute: 'max-to-show',
+			},
+			/**
+			 * @ignore
+			 */
+			mini: {
+				type: Boolean,
+				reflect: true
```

Comment:
Ditto here around using state, although then it wouldn't reflect so if that's actually needed, another mechanism would be required.

Reply:
Hmm... this seems to just get set based on the opener type, so might not even need to be a property at all.

Reply:
It does also set in `_chomp` for if the width becomes quite small. I tried removing it and it gets a bit overly complicated. I don't think we need for it to reflect, so state seems to work.